packages/back-end/src/routers/demo-datasource-project/demo-datasource-project.controller.ts

```diff
@@ -375,6 +357,28 @@ export const postDemoDatasourceProject = async (
         });
       })
     );
+    // todo: save the experiment IDs
+
+    // Create feature
+    await Promise.all(
+      DEMO_FEATURES.map(async (f, i) => {
+        const featureToCreate: FeatureInterface = {
+          ...f,
+          id: getDemoDataSourceFeatureId(),
+          project: project.id,
+          organization: org.id,
+          dateCreated: new Date(),
+          dateUpdated: new Date(),
+        };
+
+        // TODO: Ref: experiment-ref: Update this when updating demo data (features, experiments, etc.)
+        // Replaces the experimentId - Assumes the 2nd rule for all demo features is an ExperimentRefRule. If this is no longer true, we will need to change how the demo data is generated.
+        (featureToCreate.environmentSettings.production
+          .rules[1] as ExperimentRefRule).experimentId = experiments[i]?.id;
```

Comment:
This seems very fragile.  Can we loop through the rules and check if they type is `experiment-ref` instead?  Also, look through every environment instead of hard-coding `production` here.

Reply:
This was addressed [here](https://github.com/growthbook/growthbook/pull/1499#issuecomment-1688972938).